megamek/src/megamek/client/ui/swing/gameConnectionDialogs/ConnectDialog.java

```diff
@@ -0,0 +1,108 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.client.ui.swing.gameConnectionDialogs;
+
+import java.awt.BorderLayout;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import javax.swing.SwingConstants;
+
+import megamek.client.ui.Messages;
+
+/** The Connect to game (as Bot or Player) dialog */
+public class ConnectDialog extends AbstractGameConnectionDialog {
+    private static final long serialVersionUID = 5895056240077042429L;
+
+    private String serverAddress;
+    private JTextField serverAddressField;
+
+    public ConnectDialog(JFrame frame) {
+        super(frame, Messages.getString("MegaMek.ConnectDialog.title"), true);
+    }
+
+    //region Initialization
+    @Override
+    protected void initComponents() {
+        JLabel yourNameL = new JLabel(Messages.getString("MegaMek.yourNameL"), SwingConstants.RIGHT);
+        JLabel serverAddrL = new JLabel(Messages.getString("MegaMek.serverAddrL"), SwingConstants.RIGHT);
+        JLabel portL = new JLabel(Messages.getString("MegaMek.portL"), SwingConstants.RIGHT);
+        setPlayerNameField(new JTextField(getClientPreferences().getLastPlayerName(), 16));
+        getPlayerNameField().addActionListener(this);
+        serverAddressField = new JTextField(getClientPreferences().getLastConnectAddr(), 16);
+        serverAddressField.addActionListener(this);
+        setPortField(new JTextField(getClientPreferences().getLastConnectPort() + "", 4));
+        getPortField().addActionListener(this);
+        
+        JPanel middlePanel = new JPanel(new GridBagLayout());
+        GridBagConstraints c = new GridBagConstraints();
+        c.fill = GridBagConstraints.NONE;
+        c.weightx = 0.0;
+        c.weighty = 0.0;
+        c.insets = new Insets(5, 5, 5, 5);
+        c.gridwidth = 1;
+        
+        addOptionRow(middlePanel, c, yourNameL, getPlayerNameField());
+        addOptionRow(middlePanel, c, serverAddrL, serverAddressField);
+        addOptionRow(middlePanel, c, portL, getPortField());
+        
+        add(middlePanel, BorderLayout.CENTER);
+        
+        createButtons();
+        
+        pack();
+        setResizable(false);
+        center();
+    }
+    //endregion Initialization
```

Comment:
My feeling would be (not because it's important right here, more as a matter principle) that InitComponents might rather be 
"protected JPanel createMainPanel() { ... }"
Everything from "add(middlePanel..." to center() (lines 71-77) would then move to the AbstractGCD and in the subclass it would just "return middlePanel;" instead. In this way the subclass can create whatever it wants in that panel and just return it and it does not have to "remember" doing anything (e.g. to createButtons()).
Of course, this forces createbuttons() on the subclasses but I suppose that's the buttons we need for a game conection dialog...
I mean this purely as a suggestion.

Reply:
I concur, that's better code. Implement.